source/base/parameter_handler.cc

```diff
@@ -1668,36 +1675,20 @@ bool ParameterHandler::read_input (std::istream &input,
   // the last line to end in a backslash.
   if (is_concatenated)
     {
-      status &= scan_line (fully_concatenated_line, filename, current_line_n);
+      scan_line (fully_concatenated_line, filename, current_line_n);
     }
 
-  if (status && (saved_path != subsection_path))
+  if (saved_path != subsection_path)
     {
-      std::cerr << "Unbalanced 'subsection'/'end' in file <" << filename
-                << ">." << std::endl;
-      if (saved_path.size()>0)
-        {
-          std::cerr << "Path before loading input:" << std::endl;
-          for (unsigned int i=0; i<saved_path.size(); ++i)
-            std::cerr << std::setw(i*2+4) << " "
-                      << "subsection " << saved_path[i] << std::endl;
-        }
-      std::cerr << "Current path:" << std::endl;
-      for (unsigned int i=0; i<subsection_path.size(); ++i)
-        std::cerr << std::setw(i*2+4) << " "
-                  << "subsection " << subsection_path[i] << std::endl;
-
-      // restore subsection we started with and return failure:
+      // restore subsection we started with before throwing the exception:
       subsection_path = saved_path;
-      return false;
+      AssertThrow (false, ExcUnbalancedSubsections (filename));
```

Comment:
The old error message actually contained a lot more information with the set of open subsections. Would it be possible to incorporate that into the new error message somehow?